App/App.cpp

```diff
@@ -1,28 +1,48 @@
-#include <stdio.h>
-#include <climits>
-#include "App.h"
-#include "SgxException.h"
-#include "EnclaveManager.h"
-
-long affineTransformation(unsigned long x, long targetRangeMin, long targetRangeMax, unsigned long sourceRangeMin, unsigned long sourceRangeMax) {
-	return ((double)x - (double)sourceRangeMin)*((double)targetRangeMax - (double)targetRangeMin) / ((double)sourceRangeMax - (double)sourceRangeMin) + (double)targetRangeMin;
-}
-
-int generateRandom(long min, long max, long *result) {
-	try {
-		EnclaveManager enclave;
-		unsigned long generated = enclave.generateRandom();
-
-		*result = affineTransformation(generated, min, max, 0, ULONG_MAX);
-		printf("Transforming random value from SGX [0, %lu] to [%ld, %ld]: %lu -> %ld \n", ULONG_MAX, min, max, generated, *result);
-	} catch (SgxException& e) {
-		printf("%s\n", e.what());
-		return -1;
-	}
-
-	return 0;
-}
-
-int main() {
-	return 0;
-}
+#include <stdio.h>
+#include <climits>
+#include "App.h"
+#include "SgxException.h"
+#include "EnclaveManager.h"
+
+long affineTransformation(unsigned long x, long targetRangeMin, long targetRangeMax, unsigned long sourceRangeMin, unsigned long sourceRangeMax) {
+	return ((double)x - (double)sourceRangeMin)*((double)targetRangeMax - (double)targetRangeMin) / ((double)sourceRangeMax - (double)sourceRangeMin) + (double)targetRangeMin;
+}
+
+int generateRandom(long min, long max, long *result) {
+	try {
+		EnclaveManager enclave;
```

Comment:
`EnclaveManager` sounds like an object that should be created once but it's created and destroyed in `generateRandom` and in `initRemoteAttestation` functions. Maybe they should take it as an argument.

Reply:
Your point illustrutes a misdesign but not necessarily in a way you described it. It makes sense for RA to exist on its own and create a new enclave (this will be used by other Gardener functionalities in the future) , however RA should be also performed as a part of RNG. I will redesign it in that manner.